dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/PreferredName.cshtml.cs

```diff
@@ -0,0 +1,67 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace TeacherIdentity.AuthServer.Pages.SignIn.Trn;
+
+[BindProperties]
+public class PreferredName : PageModel
+{
+    private IIdentityLinkGenerator _linkGenerator;
+
+    public PreferredName(IIdentityLinkGenerator linkGenerator)
+    {
+        _linkGenerator = linkGenerator;
+    }
+
+    public string? FirstName => HttpContext.GetAuthenticationState().FirstName;
+    public string? LastName => HttpContext.GetAuthenticationState().LastName;
+
+    // Properties are set in the order that they are declared. Because the value of HasPreferredName
+    // is used in the conditional RequiredIfTrue attribute, it should be set first.
+    [Display(Name = " ")]
```

Comment:
Wasn't able to set name to required page title as this string is evaluated at compile time. Apparently adding a space works though!